crates/interfaces/src/provider.rs

```diff
@@ -88,25 +88,25 @@ pub enum ProviderError {
     #[error("merkle trie root mismatch at #{block_number} ({block_hash}): got {got}, expected {expected}")]
     StateRootMismatch {
         /// The expected root.
-        expected: B256,
+        expected: Box<B256>,
         /// The calculated root.
-        got: B256,
+        got: Box<B256>,
         /// The block number.
         block_number: BlockNumber,
         /// The block hash.
-        block_hash: BlockHash,
+        block_hash: Box<BlockHash>,
```

Comment:
could we convert this into a struct and box that instead?

Reply:
Yeah, good idea. I'll do that for `got, expected` as well